AEPMedia/Sources/MediaCollection/MediaCollectionReportHelper.swift

```diff
@@ -41,19 +43,39 @@ class MediaCollectionReportHelper {
     /// - host: Host component of the URL
     /// - sessionId: the session id of the Media session
     /// - Returns: URL for sending media hits
-    static func getTrackingURLForEvents(host: String, sessionId: String?) -> URL? {
-        guard !host.isEmpty, let sessionId = sessionId, !sessionId.isEmpty else {
+    static func getTrackingURLForEvents(trackingServer: String, sessionId: String?) -> URL? {
+        guard !trackingServer.isEmpty, let sessionId = sessionId, !sessionId.isEmpty else {
             Log.warning(label: LOG_TAG, "[\(Self.CLASS_NAME)<\(#function)>] - Unable to create tracking url for events. Host (MediaCollectionServer) url or collection server sessionId is empty.")
             return nil
         }
 
+        let serverAddress = getHostAndPath(trackingServer)
+
         var urlComponents = URLComponents()
         urlComponents.scheme = "https"
-        urlComponents.host = host
-        urlComponents.path = "/api/v1/sessions/\(sessionId)/events"
+        urlComponents.host = serverAddress["host"]
+        urlComponents.path = (serverAddress["path"] ?? "") + "/api/v1/sessions/\(sessionId)/events"
         return urlComponents.url
     }
 
+    static func getHostAndPath(_ trackingServer: String) -> [String: String] {
+        var ret =  [String: String]()
+        let trackingServerComponents = trackingServer.components(separatedBy: "/")
+
+        ret["host"] = trackingServerComponents[0]
+
+        var path = ""
+        if trackingServerComponents.count > 1 {
+            for i in 1...trackingServerComponents.count-1 {
+                path += "/" + trackingServerComponents[i]
+            }
+        }
```

Comment:
Would this work?
```
if (trackingServerComponents.count > 1 {
    trackingServerComponents.removeFirst(1)
    path = trackingServerComponents.joined(separator: "/")
}
```

Reply:
I can try this.

Reply:
It was not straightforward to solve my negative test case where we want to pass the values without any modifications from the customer input.

This was failing the happy test case. Was getting one extra "/" when path was empty.
```
 let path = "/" + components.joined(separator: "/")
 ```

and this was failing the negative test case ("//abc.com/va//"). Was getting one less "/"
```
 let path =  components.joined(separator: "/")
```

Reply:
Maybe something like
`let path = components.isEmpty ? "" : "/" + components.joined(separator: "/")`

Reply:
oh yes that would work.